src/main/java/games/strategy/engine/GameEngineUtils.java

```diff
@@ -0,0 +1,43 @@
+package games.strategy.engine;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import games.strategy.util.Version;
+
+/**
+ * A collection of methods for use with game engines.
+ */
+public final class GameEngineUtils {
+  private GameEngineUtils() {}
+
+  /**
+   * Indicates the engines with the specified versions are compatible.
+   *
+   * @param engineVersion The first engine version.
+   * @param otherEngineVersion The second engine version.
+   *
+   * @return {@code true} if the engines are compatible; otherwise {@code false}.
+   */
+  public static boolean isEngineCompatibleWithEngine(final Version engineVersion, final Version otherEngineVersion) {
+    checkNotNull(engineVersion);
+    checkNotNull(otherEngineVersion);
+
+    return engineVersion.compareTo(otherEngineVersion, Version.CompareOption.IGNORE_MICRO) == 0;
```

Comment:
I like the idea of extracting this kind of logic to a dedicated class, but in my opinion this doesn't go far enough:
I don't think such a single special case should be reason enough to handle them from within the Version class.
If we replaced this return statement with 3 Version#getMajor Minor and Point equals checks, we could drop all of the ignoreMicro logic and achieve the exact same result.

Reply:
I tend to agree with you, and, in fact, I briefly considered doing something similar to this before pushing the previous commit, but I felt that the amount of duplicate logic in the `isEngineCompatibleWithMap()` implementation would have been a smell.  However, that was before I introduced your suggested change to use `Comparator#comparingInt()` et. al.  Maybe that will clean it up a bit.  Let me give it another iteration to see if that works.

(Also, since we're talking about logic that doesn't belong in `Version` proper, I'm wondering if we should revisit how we handle the `dev` qualifier.  Now that I've been mucking with this class, it seems like something a bit too specific for this general-purpose type.  But let's save that for another PR. :smile:)